### ISSUE TITLE
BagFile.getMessageOnTopicAtIndex throws IndexOutOfBoundsException rather than ArrayIndexOutOfBoundsException 

### ISSUE THREAD

**Issue by tobidelbruck** [swri-robotics/bag-reader-java#31]
Could it be that line 580 of  https://github.com/swri-robotics/bag-reader-java/blob/master/src/main/java/com/github/swrirobotics/bags/reader/BagFile.java should be changed from
```
 if (index > indexes.size()) {
            throw new ArrayIndexOutOfBoundsException(index);
        }
```
to

 ```
  if (index >= indexes.size()) {
            throw new ArrayIndexOutOfBoundsException(index);
        }
```
i.e. change > to >= 

I think that is why I'm getting IndexOutOfBounds rather than ArrayIndexOutOfBounds exception here.
I suppose I should fix it and submit a pull request, which will be a first for me.

**Comment by pjreed** [comment]
Good catch, that is definitely a bug.  A PR would be welcome. :smile:

**Comment by pjreed** [comment]
Fixed in #33.
